krummstab/__init__.py

```diff
@@ -228,16 +240,38 @@ def verify_sheet_root_dir() -> None:
         logging.critical("The given sheet directory is not valid!")
 
 
+def get_submission_info(path: pathlib.Path) -> dict:
+    """
+    Load the submission info of the team directory. In particular the
+    team and if it is a relevant team.
+    """
+    try:
+        with open(
+                path / SUBMISSION_INFO_FILE_NAME, "r", encoding="utf-8"
+        ) as submission_info_file:
+            submission_info = json.load(submission_info_file)
+            submission_info_schema = json.loads(
+                resources.files(schemas).joinpath("submission-info-schema.json").read_text(encoding="utf-8"))
+            jsonschema.validate(submission_info, submission_info_schema, jsonschema.Draft7Validator)
+            return submission_info
+    except FileNotFoundError:
+        logging.critical("The submission.json file does not exist.")
+    except NotADirectoryError:
+        logging.critical(f"The path '{path}' is not a team directory.")
+    except jsonschema.exceptions.ValidationError:
+        logging.critical("The submission.json file has not the right format.")
+
+
 def get_all_team_dirs() -> Iterator[pathlib.Path]:
     """
-    Return all team directories within the sheet root directory. It is assumed
-    that all team directory names start with some digits, followed by an
-    underscore, followed by more characters. In particular this excludes
-    other directories that may be created in the sheet root directory, such as
+    Return all team directories within the sheet root directory. To find the team
+    directories, the submission.json files are used. This excludes other
+    directories that may be created in the sheet root directory, such as
     one containing combined feedback.
     """
     for team_dir in args.sheet_root_dir.iterdir():
-        if team_dir.is_dir() and re.match(r"[0-9]+_.+", team_dir.name):
+        if team_dir.is_dir() and team_dir != args.sheet_root_dir / COMBINED_DIR_NAME\
+                and get_submission_info(team_dir):
```

Comment:
`get_submission_info` cannot fail, so you don't have to test for it.

Reply:
The test `team_dir != args.sheet_root_dir / COMBINED_DIR_NAME` looks fishy to me: isn't `COMBINED_DIR_NAME` something like `feedback_collected` and `team_dir` something like `team12345_X-Y`?